### PR TITLE
[AMORO-2198] Support filters of Snapshots page for table

### LINKED ISSUE #2198: [Subtask]: [Dashboard] Support filters of `Snapshots` page for table
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

Support filters for `Snapshots` page. As desceibed in #2034 , we are going to support two filters in the `Snapshots` page.

image

**Branches/Tags Filter:**
- the tags/branches in the list are from the HTTP API introduced from https://github.com/NetEase/amoro/pull/2172
- the HTTP API will return at least 1 branch
- the first branch of the list should be the default value of the filter
- only support select 1 branch or tag at a time

**Operations Filter:**
- the list contains 3 fixed items, `all`,`optimizing`,`non-optimizing`
- `all` is the default value
- after swith a tag/branch, the operations filter should change to select the default value `all`
- only support select 1 operation at a time

Both filters will task effects on the HTTP API `.../tables/{table}/snapshots` in the format of: 
`.../tables/{table}/snapshots?ref={tag/branch}&operation={all}`

Besides, we also have to support displying how many branches and tags there are, which also comes from this HTTP API
- https://github.com/NetEase/amoro/pull/2172.

### Parent issue

- #1354

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by minteliuwm**
## Why are the changes needed?

Close #2198.

## Brief change log

- Support filters of Snapshots page for table

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by wangtaohz — COMMENTED** [review]
Thanks for you contribution! 

I test it locally, and it works well.

image
image
image

**Comment by wangtaohz** [comment]
There is a tiny problem, when I select a `tag`, the icon is still a `branch`.

image

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java [hunk 1]
```diff
@@ -647,6 +648,7 @@ private List<TagOrBranchInfo> getTableTagsOrBranches(
               result.add(new TagOrBranchInfo(name, snapshotRef));
             }
           });
+      result.sort(Comparator.comparing(TagOrBranchInfo::getName));
```
Comment (wangtaohz) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java:651]: Notes: 
Branches and Tags of Iceberg don't have a create time, and the order is unstable, so I sort them by name to make it easier for users to view.

**Comment by wangtaohz** [comment]
When there are many branches/tags, maybe we need a scrollbar. @minteliuwm 

image

**Review by wangtaohz — APPROVED** [review]
LGTM

**Comment by wangtaohz** [comment]
Locally test result
image

table with no snapshot
image

**Review by baiyangtx — APPROVED** [review]
LGTM

**Pull request merged**